packages/test/test-end-to-end-tests/src/test/gc/gcStateResetInSummaries.spec.ts

```diff
@@ -0,0 +1,276 @@
+            assert(gcSummaryDetails.unrefTimestamp === undefined, `DataStore ${id} shouldn't have unref timestamp`);
+            assert.deepStrictEqual(gcSummaryDetails.usedRoutes, [""], `DataStore ${id} should only have self route`);
+        }
+    }
+
+    before(function() {
+        provider = getTestObjectProvider();
+        // These tests validate the end-to-end behavior of summaries when GC is enabled / disabled. This behavior
+        // is not affected by the service. So, it doesn't need to run against real services.
+        if (provider.driver.type !== "local") {
+            this.skip();
+        }
+    });
+
+    beforeEach(async () => {
+        // Wrap the document service factory in the driver so that the `uploadSummaryCb` function is called every
+        // time the summarizer client uploads a summary.
+        (provider as any)._documentServiceFactory = wrapDocumentServiceFactory(
+            provider.documentServiceFactory,
+            uploadSummaryCb,
+        );
+    });
+
+    afterEach(() => {
+        latestAckedSummary = undefined;
+        latestSummaryContext = undefined;
+        latestUploadedSummary = undefined;
+    });
+
+    it("removes GC state and marks all objects as referenced on disabling GC", async () => {
+        // Create a document with GC enabled.
+        mainContainer = await createContainer(true /* gcAllowed */);
+        const mainDataStore = await requestFluidObject<TestDataObject>(mainContainer, "default");
+        await provider.ensureSynchronized();
+
+        const summarizerClient = await getNewSummarizer(false /* disableGC */);
+
+        // Mark the second data store as referenced by storing its handle in a referenced DDS.
+        const newDataStore = await dataObjectFactory.createInstance(mainDataStore.containerRuntime);
+        mainDataStore._root.set("newDataStore", newDataStore.handle);
+
+        // Validate that GC ran.
+        await validateGCRan(summarizerClient);
+
+        // Mark the data store as unreferenced by deleting its handle from the DDS. Validate that GC ran and the
+        // unreferenced data store is marked as such in GC state.
+        mainDataStore._root.delete("newDataStore");
+        await validateGCRan(summarizerClient, newDataStore.id);
+
+        // Load a new summarizer from the last summary with GC disabled. Validate that GC did not run and that
+        // previous GC state is removed expect for used routes which marks all data stores as referenced.
+        assert(latestAckedSummary !== undefined, "Summary ack isn't available as expected");
+        const summarizerClient2 = await getNewSummarizer(
+            true /* disableGC */,
+            undefined /* gcAllowed */,
+            latestAckedSummary.summaryAck.contents.handle,
+        );
+        await validateGCDidNotRun(summarizerClient2);
+
+    // This test has increased timeout because it waits for multiple summaries to be uploaded to server. It then also
```

Comment:
Where is the increased timeout configured?

Reply:
I removed the timeouts since these test only run against local server. But  forgot to remove the comment.